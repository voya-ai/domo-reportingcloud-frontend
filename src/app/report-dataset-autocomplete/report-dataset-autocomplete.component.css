.mat-spinner {
  height: 20px;
  display: inline-block;
  width: 20px;
  left: -28px;
  top: 6px;
  position: relative;
}
