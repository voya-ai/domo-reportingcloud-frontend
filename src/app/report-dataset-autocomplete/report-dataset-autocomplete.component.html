<mat-form-field>
  <input
    matInput
    placeholder="Dataset"
    [defaultValue]="dataSet"
    [matAutocomplete]="autoDataset"
    [formControl]="datasetControl">
</mat-form-field>

<mat-spinner *ngIf="datasetLoading"></mat-spinner>

<mat-autocomplete
  #autoDataset="matAutocomplete"
  [displayWith]="displayDataset">
  <mat-option
    *ngFor="let option of filteredDatasetOptions"
    [value]="option"
    (onSelectionChange)="selectionChanged(option)">
     {{ option.name }}
  </mat-option>
</mat-autocomplete>
