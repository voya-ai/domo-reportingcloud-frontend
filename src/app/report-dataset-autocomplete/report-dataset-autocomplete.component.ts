import { Component, Input, Output, EventEmitter, OnInit } from '@angular/core';
import { FormControl } from '@angular/forms';

import { DatasetService } from 'app/dataset.service';

@Component({
  selector: 'app-report-dataset-autocomplete',
  templateUrl: './report-dataset-autocomplete.component.html',
  styleUrls: ['./report-dataset-autocomplete.component.css']
})
export class ReportDatasetAutocompleteComponent implements OnInit {
  datasetControl = new FormControl();
  @Input() dataSet: any;
  @Output() onSelectionChange: EventEmitter<any> = new EventEmitter<any>();
  filteredDatasetOptions: string[];
  datasetLoading: boolean = false;

  constructor(private _datasetService: DatasetService) {}

  ngOnInit() {
    this.datasetControl
      .valueChanges
      .debounceTime(400)
      .distinctUntilChanged()
      .subscribe((val : string) => {
        this.datasetLoading = true;
        this._datasetService.search(val)
        .toPromise()
        .then(function(result : Array<any>) {
          this.datasetLoading = false;
          this.filteredDatasetOptions = result;
        }.bind(this));
      });

    if(this.dataSet) {
      this.datasetLoading = true;
      this._datasetService.get(this.dataSet)
        .subscribe((dataset: any) => {
          this.datasetLoading = false;
          this.datasetControl.setValue(dataset);
          this.selectionChanged(dataset);
        });
    }
  }

  selectionChanged(option: any) {
    this.onSelectionChange.emit(option);
  }

  displayDataset(dataset : any) : string {
    return dataset ? dataset.name : dataset;
  }

}
