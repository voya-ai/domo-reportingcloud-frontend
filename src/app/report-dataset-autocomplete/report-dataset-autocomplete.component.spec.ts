import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { Observable } from 'rxjs/Observable';
import {
  MatAutocompleteModule
} from '@angular/material';

import { DatasetService } from 'app/dataset.service';

import 'rxjs/add/observable/of';

import { ReportDatasetAutocompleteComponent } from './report-dataset-autocomplete.component';

class MockDatasetService {
  search(val : string) : Observable<any[]> {
    return Observable.of<any[]>([]);
  }
}

describe('ReportDatasetAutocompleteComponent', () => {
  let component: ReportDatasetAutocompleteComponent;
  let fixture: ComponentFixture<ReportDatasetAutocompleteComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      providers: [
        { provide: DatasetService, useClass: MockDatasetService }
      ],
      declarations: [ ReportDatasetAutocompleteComponent ],
      imports: [
        MatAutocompleteModule,
        FormsModule,
        ReactiveFormsModule
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ReportDatasetAutocompleteComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
