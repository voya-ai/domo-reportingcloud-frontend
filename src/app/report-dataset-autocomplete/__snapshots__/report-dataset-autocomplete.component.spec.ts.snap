// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDatasetAutocompleteComponent should match the snapshot 1`] = `
<app-report-dataset-autocomplete
  _datasetService={[Function MockDatasetService]}
  datasetControl={[Function FormControl]}
  datasetLoading="false"
  onSelectionChange={[Function EventEmitter]}
>
  <mat-form-field>
    
    
    <input
      aria-autocomplete="list"
      aria-expanded="false"
      aria-owns="mat-autocomplete-1"
      autocomplete="off"
      class="ng-untouched ng-pristine ng-valid"
      matinput=""
      ng-reflect-autocomplete="[object Object]"
      ng-reflect-form="[object Object]"
      placeholder="Dataset"
      role="combobox"
      value="undefined"
    />
    
  
  </mat-form-field>
  
  
  
  <mat-autocomplete
    class="mat-autocomplete"
    ng-reflect-display-with="function (dataset) {
          r"
  >
    
  </mat-autocomplete>
  
</app-report-dataset-autocomplete>
`;
